### ISSUE TITLE
Feature Request: Release Envoy v1.21.1 

### ISSUE THREAD

**Issue by clouddev-code** [aws/aws-app-mesh-roadmap#393]
The Envoy team released v1.21.1: https://github.com/envoyproxy/envoy/releases/tag/v1.21.1

**Are you currently working around this issue?**
We should update the App Mesh Envoy Container Image to v1.21.1.

**Comment by suniltheta** [comment]
The images have been released:

* For `me-south-1`:
  ```
  772975370895.dkr.ecr.me-south-1.amazonaws.com/aws-appmesh-envoy:v1.21.1.0-prod
  ```
* For `ap-east-1`:
  ```
  856666278305.dkr.ecr.ap-east-1.amazonaws.com/aws-appmesh-envoy:v1.21.1.0-prod
  ```
* For `eu-south-1`:
  ```
  422531588944.dkr.ecr.eu-south-1.amazonaws.com/aws-appmesh-envoy:v1.21.1.0-prod
  ```
* For `af-south-1`:
  ```
  924023996002.dkr.ecr.af-south-1.amazonaws.com/aws-appmesh-envoy:v1.21.1.0-prod
  ```
* For `cn-north-1`:
  ```
  919366029133.dkr.ecr.cn-north-1.amazonaws.com.cn/aws-appmesh-envoy:v1.21.1.0-prod
  ```
* For `cn-northwest-1`:
  ```
  919830735681.dkr.ecr.cn-northwest-1.amazonaws.com.cn/aws-appmesh-envoy:v1.21.1.0-prod
  ```

* For all other [regions where App Mesh is available](https://docs.aws.amazon.com/general/latest/gr/appmesh.html):
  ```
  840364872350.dkr.ecr.<region>.amazonaws.com/aws-appmesh-envoy:v1.21.1.0-prod
  ```

  e.g.
  ```
  840364872350.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.21.1.0-prod
  ```
* Public ECR Image:
  ```
  public.ecr.aws/appmesh/aws-appmesh-envoy:v1.21.1.0-prod
  ```  

We'll keep this issue open as we update the recommended image our other projects (e.g. aws/aws-app-mesh-controller-for-k8s), examples, and documentation.

**Comment by suniltheta** [comment]
Thanks to Envoy community, this addresses the following CVE(s):


* [CVE-2021-43824](https://github.com/envoyproxy/envoy/security/advisories/GHSA-vj5m-rch8-5r2p) (CVSS Score 6.5, Medium): Envoy 1.21.0 and earlier - Potential null pointer dereference when using JWT filter safe_regex match


* [CVE-2021-43825](https://github.com/envoyproxy/envoy/security/advisories/GHSA-h69p-g6xg-mhhh) (CVSS Score 6.1, Medium): Envoy 1.21.0 and earlier - Use-after-free when response filters increase response data, and increased data exceeds downstream buffer limits.


* [CVE-2021-43826](https://github.com/envoyproxy/envoy/security/advisories/GHSA-cmx3-fvgf-83mf) (CVSS Score 6.1, Medium): Envoy 1.21.0 and earlier - Use-after-free when tunneling TCP over HTTP, if downstream disconnects during upstream connection establishment


* [CVE-2022-21654](https://github.com/envoyproxy/envoy/security/advisories/GHSA-5j4x-g36v-m283) (CVSS Score 7.3, High): Envoy 1.7.0 and later - Incorrect configuration handling allows mTLS session re-use without re-validation after validation settings have changed.


* [CVE-2022-21655](https://github.com/envoyproxy/envoy/security/advisories/GHSA-7r5p-7fmh-jxpg) (CVSS Score 7.5, High): Envoy 1.21 and earlier - Incorrect handling of internal redirects to routes with a direct response entry


* [CVE-2022-23606](https://github.com/envoyproxy/envoy/security/advisories/GHSA-9vp2-4cp7-vvxf) (CVSS Score 4.4, Moderate): Envoy 1.20 and later - Stack exhaustion when a cluster is deleted via Cluster Discovery Service.

 

* [CVE-2022-21656](https://github.com/envoyproxy/envoy/security/advisories/GHSA-c9g7-xwcv-pjx2) (CVSS Score 3.1, Low): Envoy 1.20.1 and earlier - X.509 subjectAltName matching (and nameConstraints) bypass. IMPORTANT: Due to significant divergence in affected source code between Envoy versions 1.20 and earlier branches, it is not feasible to backport the fix for CVE-2022-21656 into the 1.19 or 1.18 stable branches without increasing the risk of destabilizing it. As such it will only be fixed in 1.20.


* [CVE-2022-21657](https://github.com/envoyproxy/envoy/security/advisories/GHSA-837m-wjrv-vm5g) (CVSS Score 3.1, Low): Envoy 1.20.1 and earlier - X.509 Extended Key Usage and Trust Purposes bypass

**Comment by suniltheta** [comment]
We recently discovered that in this Image there is a bug that breaks the integration with Jaeger and Datadog agent https://github.com/aws/aws-app-mesh-roadmap/issues/401 

We will follow up this with a new patch image release.
